arwen/contexts/output.go

```diff
@@ -301,6 +301,18 @@ func (context *outputContext) GetVMOutput() *vmcommon.VMOutput {
 
 	context.removeNonUpdatedCode(context.outputState)
 
+	gasUsed := uint64(0)
```

Comment:
take gas check function into a separate function. These checks are not enough -> we need to iterate over all transfer in an output account and accumulate gasLocked + gasLimit from there as well.

Reply:
Done.

Reply:
looks good. add safemath + necessary changes in accumulated gas function